### DISCUSSION TITLE
Ensoniq ESQ-1 adaptation

### DISCUSSION THREAD

**Discussion by Mostelin** [christofmuc/KnobKraft-orm#119] [General]
I have created an Ensoniq ESQ-1 adaptation (#120), which should also operate with the Ensoniq ESQ-M and Ensoniq SQ-80. I have also created a [page ](https://github.com/christofmuc/KnobKraft-orm/wiki/Ensoniq-ESQ-1) for this in the Wiki.

**Comment by markusschloesser** [comment]
Hi,
First of all thank you for writing this! :-)
I downloaded your adaptation with the intention to use it with the SQ8L plugin, which can receive ESQ-1 and SQ80 sysex files (i know, edge case 🤪😇)

2 things: 
1. the knobkraft complains about the file name 
`Adaptation: Warning: file name ensoniqesq1.88 is not a valid module identifier in Python, please use only lower case letters and numbers
Adaptation: Failure loading python module: ModuleNotFoundError: No module named 'ensoniqesq1'
Unloading adaptation module C:\Users\Markus\KnobKraft-Adaptations\ensoniqesq1.88.py`
Deleting the "." fixes that.
2. Currently stroking my beard as to how to connect knobkraft and your adaptation with the plugin in Ableton. Will probably involve some weird virtual midi connections, or am I completely thinking to strangely?

**Reply by Mostelin** [reply]
Markus
I should have added a note that the version number (.88) must be removed from the file name before use. I've done that now. Thanks for bringing this back to my attention. I will not do things the same way next time :)

I'm flattered that you think I have something useful to say about working with the SQ8L. I also have this amazing plugin, but I have no experience of interfacing it to anything else. I've only ever fed standard Ensoniq bank files of 40 programs into it, and occasionally saved them using the SQ8L's own file format, which I think permits 128 programs in one file. So my use has been only file-based.

My use of SQ8L is about to diminish, as I have the Arturia SQ80 V too now, which appears capable of saving any number of programs in a single file, and is a bit more stable on my set-up. In fact, it was the SQ80 V that spurred this recent interest in getting all available ESQ/SQ programs cleaned up and categorised. It is a shame about the SQ80 V import bug that misread three of the four LFO waveforms. They released an update about a week ago, but it fixes only two of the misreadings. Now, SAW is translated to Ramp, an inversion of the SAW profile.

**Reply by christofmuc** [reply]
Very good question on the MIDI communication with the plugin. I have yet to encounter a VST that actually implements sysex transfers and communication over MIDI/virtual MIDI. I think they all only allow file import/export, but no realtime communication. Maybe I am wrong?

**Reply by markusschloesser** [reply]
I know at least of the SQ8L (has a specific menu entry for this) and Dexed  😊
Extra detour is to be taken with the SQ8L because it is 32bit only, so I have it hosted in ddmf's metaplugin.
Of course I could simply use a file based import, but it is rather cumbersome IMHO.

**Reply by christofmuc** [reply]
@markusschloesser No kidding? This minute I was looking at the Dexed source code and concluded it didn't do it. But then, there's some more code. Sounds like I have to dig deeper, but tonight I'm through. Off to bed.

**Reply by markusschloesser** [reply]
https://github.com/asb2m10/dexed/wiki/Dexed-as-a-DX7-editor :-)

**Reply by markusschloesser** [reply]
I imported around 4.200 presets into Knobkraft and using LoopBe as an internal midi cable I can (after clicking "Receive Program" in the GUI), receive sysex presets from Knobkraft! Very cool!! :-)

2 downsides:
1. I have to click "Receive Program" EVERY fucking time
2. I then have to select the midi port every time as well
LoopBe is multi-client capable and goes "around" Ableton

**Comment by Mostelin** [comment]
I am testing a revision of the ESQ-1 adaptation that observes the peculiar Ensoniq mix of ASCII and non-ASCII encoding of program names. The advantage of this is that a name displayed in the ORM will more closely resemble the name as it appears in the synth's display. For example, ECHO1. is a factory default program in the ESQ-1 and appears to have six characters in its name, but a straight ASCII translation of it results in ECHO#, which has only five. The 1. is encoded as a single character and occupies a single character space in the display. This oddity, incidentally, is what makes translation the other way (computer to synth) tricky, as 1. can obviously be encoded in two different ways.

**Reply by christofmuc** [reply]
Yes, but it should be possible to do. The Oberheim Matrix 1000 does the same, it can represent characters in ASCII or in lower values that are special characters in ASCII, so there is no guaranteed back and forth of patch to name and v.v. But if the fingerprinting makes sure the patch name is ignored that doesn't cause any hiccups.

**Comment by christofmuc** [comment]
Closing this as done.
